compiler/rustc_data_structures/src/sharded.rs

```diff
@@ -64,6 +64,41 @@ impl<T> Sharded<T> {
     }
 }
 
+#[derive(Clone)]
+pub struct SSharded<T, L: SLock> {
```

Comment:
Can we use a more descriptive name, or at least put a comment of some kind on this? It looks very similar to Sharded.

Reply:
Thanks for the reviewing, that makes sense! The current changes are just to locate the causes of the backwards and use some rudimentary methods to try and fix them.